pkgs/top-level/impure.nix

```diff
@@ -72,15 +72,24 @@ in
         else overlays homeOverlaysDir
       else []
 
+  # An impurity in NixOS to pass on when invoking pkgs.nixos,
+  # pkgs.nixosTest, pkgs.nixosTests.
+, nixosExtraModules ? let e = builtins.getEnv "NIXOS_EXTRA_MODULE_PATH";
```

Comment:
I would maybe call this `nixosTestExtraModules` so it doesn't just sound like some layer violation.